network/dag/state.go

```diff
@@ -217,15 +222,19 @@ func (s *state) Add(ctx context.Context, transaction Transaction, payload []byte
 	}), stoabs.WithWriteLock())
 }
 
-func (s *state) updateTrees(tx stoabs.WriteTx, transaction Transaction) error {
+func (s *state) updateState(tx stoabs.WriteTx, transaction Transaction) error {
+	if s.highestLC.get() < transaction.Clock() {
```

Comment:
This is the point where it'll break in the future. This works because the write tx has an exclusive lock on the DB. As soon as parallel write TXs are allowed (not anytime soon) you'll need a `setIfBigger` operation on the `highestLC`.

Reply:
added + tests